test/unit/master/test_build.py

```diff
@@ -445,6 +445,13 @@ def test_delete_temporary_build_artifact_files_skips_results_tarball(self):
 
         mock_shutil.rmtree.assert_called_once_with(expected_async_delete_call_path, ignore_errors=True)
 
+    def test_exception_during_postbuild_tasks_fails_build(self):
+        self.mock_util.fs.zip_directory.side_effect = FileNotFoundError('Where my files at?')
+
+        build = self._create_test_build(BuildStatus.FINISHED)
+
+        self.assertEqual(build._status(), BuildStatus.ERROR)
```

Comment:
You will say, "Whoa, why are you accessing a private method?!" And I will say, "Yes, I do not love that but all the tests in here do it and fixing it is outside the scope of this change."